tests/test_copy_novaseqx_flow_cell.py

```diff
@@ -0,0 +1,126 @@
+
+from cg.cli.demultiplex.demux import (
+    copy_flow_cell_analysis_data,
+    is_ready_for_post_processing,
+)
+from cg.constants.demultiplexing import DemultiplexingDirsAndFiles
+
+
+def test_flow_cell_is_ready_for_post_processing(
+    novaseqx_flow_cell_dir: Path, demultiplexed_flow_cell_run_directory: Path
+):
+    # GIVEN a flow cell which is ready for post processing
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(
+        novaseqx_flow_cell_dir, demultiplexed_flow_cell_run_directory
+    )
+
+    # THEN the flow cell is ready
+    assert ready
+
+
+def test_is_not_ready_without_analysis(
+    novaseqx_flow_cell_analysis_incomplete: Path, demultiplexed_flow_cell_run_directory: Path
+):
+    # GIVEN a flow cell for which analysis is not completed
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(
+        novaseqx_flow_cell_analysis_incomplete, demultiplexed_flow_cell_run_directory
+    )
+
+    # THEN it is not ready
+    assert not ready
+
+
+def test_flow_cell_is_not_demultiplexed(
+    demultiplex_not_complete_novaseqx_flow_cell: Path, demultiplexed_flow_cell_run_directory: Path
+):
+    # GIVEN a flow cell for which demultiplexing is not completed
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(
+        demultiplex_not_complete_novaseqx_flow_cell, demultiplexed_flow_cell_run_directory
+    )
+
+    # THEN it is not ready
+    assert not ready
+
+
+def test_previously_post_processed_flow_cell_is_not_ready(
+    post_processed_novaseqx_flow_cell: Path, demultiplexed_flow_cell_run_directory: Path
+):
+    # GIVEN a flow cell for which post processing is done
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(
+        post_processed_novaseqx_flow_cell, demultiplexed_flow_cell_run_directory
+    )
+
+    # THEN the flow cell is not ready
+    assert not ready
+
+
+def test_previously_copied_flow_cell_is_not_ready(
+    novaseqx_flow_cell_dir: Path, demultiplexed_flow_cell_run_directory: Path
+):
+    # GIVEN a flow cell which already exists in demultiplexed runs
+    Path(demultiplexed_flow_cell_run_directory, novaseqx_flow_cell_dir.name).mkdir()
+
+    # WHEN checking if the flow cell is ready for post processing
+    ready = is_ready_for_post_processing(
+        novaseqx_flow_cell_dir, demultiplexed_flow_cell_run_directory
+    )
+
+    # THEN the flow cell is not ready
+    assert not ready
+
+
+def test_get_latest_analysis_version_path(
+    novaseqx_flow_cell_dir: Path,
+    novaseqx_latest_analysis_version: str,
+):
+    # GIVEN a flow cell which is ready to be post processed
+
+    # WHEN extracting the latest analysis version path
+    analysis: Path = get_latest_analysis_directory(novaseqx_flow_cell_dir)
```

Comment:
```suggestion
    analysis: Path = get_latest_analysis_path(novaseqx_flow_cell_dir)
```
Given the return type hint and what is returned (object not str), name of test and comment

Reply:
fixed